.github/workflows/deploy-preview.yml

```diff
@@ -0,0 +1,194 @@
+name: deploy-preview
+
+on:
+  workflow_call:
+    inputs:
+      sha:
+        required: true
+        type: string
+      branch:
+        required: true
+        type: string
+      event_number:
+        required: true
+        type: string
+      title:
+        required: true
+        type: string
+      repo:
+        required: true
+        type: string
+      source_directory:
+        default: docs/sources
+        type: string
+      website_directory:
+        required: true
+        type: string
+      relative_prefix:
+        required: true
+        type: string
+
+env:
+  CLOUD_RUN_REGION: us-south1
+
+permissions:
+  contents: read
+  id-token: write
+
+concurrency:
+  group: ${{ github.workflow }}-${{ inputs.repo }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  deploy-preview:
+    permissions: write-all
+    runs-on: ubuntu-latest
+    steps:
+      - name: Find comment
+        uses: peter-evans/find-comment@3eae4d37986fb5a8592848f6a574fdf654e61f9e # v3.1.0
+        id: fc
+        with:
+          issue-number: ${{ inputs.event_number }}
+          comment-author: "github-actions[bot]"
+          body-includes: Deploy preview available
+
+      - name: Update comment with in-progress deploy preview
+        if: steps.fc.outputs.comment-id != '' && github.event.action != 'closed'
+        uses: peter-evans/create-or-update-comment@71345be0265236311c031f5c7866368bd1eff043 # v4.0.0
+        with:
+          comment-id: ${{ steps.fc.outputs.comment-id }}
+          issue-number: ${{ inputs.event_number }}
+          edit-mode: append
+          append-separator: newline
+          body: |
+            :building_construction: Updating deploy preview...
+
+      - uses: actions/checkout@v4
+        if: github.event.action != 'closed'
+
+      # get the Dockerfile and nginx conf
+      - name: Sparse checkout
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        if: github.event.action != 'closed'
+        with:
+          repository: "grafana/writers-toolkit"
+          ref: "robbymilo/deploy-preview"
+          path: deploy-preview-files
+          sparse-checkout: |
+            deploy-preview
+
+      - name: Build website
+        if: github.event.action != 'closed'
+        shell: bash
+        run: |
+          mkdir -p ${PWD}/dist
+          docker run -v ${PWD}/dist:/hugo/dist -v "${PWD}/${{ inputs.source_directory }}:/hugo/${{ inputs.website_directory }}" --rm grafana/docs-base:latest /bin/bash -c 'HUGO_SSI=false hugo --environment=docs --destination=dist/ --baseURL= --minify'
+          printf "%s" "add_header 'Build' '"${{ inputs.sha }}"';" > build.conf
+
+      - uses: google-github-actions/auth@6fc4af4b145ae7821d527454aa9bd537d1f2dc5f # v2.1.7
```

Comment:
Couldn't you use https://github.com/grafana/shared-workflows/tree/main/actions/push-to-gar-docker for this and the following steps without requiring a custom service account?

Reply:
push-to-gar uses login-to-gar, which then uses the service account from the `construct-service-account` step 

The service account in login-to-gar (https://github.com/grafana/shared-workflows/blob/main/actions/login-to-gar/action.yaml#L27), which is called from `push-to-gar`, is hard coded from the output of the `construct-service-account` step.

Reply:
Happy to explore this in a follow up. I don't mind that the same service account handles all the aspects of this service.

Reply:
The service account in the shared-workflow is currently just an implementation detail and IIRC @dsotirakis is investigating if we can completely get rid of it 🙂 That was the main reason for me asking about this here but it's absolutely not required 🙂

Reply:
> The service account in the shared-workflow is currently just an implementation detail and IIRC @dsotirakis is investigating if we can completely get rid of it 🙂 That was the main reason for me asking about this here but it's absolutely not required 🙂

Good to know! I'm subscribed to all activity on the grafana/shared-workflows repository so if something moves there, we'll know :) 

Thanks for clarifying

Reply:
Happy to come back and update if we can get rid of it!